prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -322,6 +354,112 @@ public APIResult validate(
+            throw FalconWebException.newAPIException(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @GET
+    @Path("describe/{extension-name}")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String getExtensionDescription(
+            @PathParam("extension-name") String extensionName) {
+        validateExtensionName(extensionName);
+        try {
+            return getResource(extensionName, README);
+        } catch (Throwable e) {
+            throw FalconWebException.newAPIException(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @GET
+    @Path("definition/{extension-name}")
+    @Produces({MediaType.APPLICATION_JSON})
+    public String getExtensionDefinition(
+            @PathParam("extension-name") String extensionName) {
+        validateExtensionName(extensionName);
+        try {
+            return getResource(extensionName,
+                    extensionName.toLowerCase() + EXTENSION_PROPERTY_JSON_SUFFIX);
+        } catch (Throwable e) {
+            throw FalconWebException.newAPIException(e, Response.Status.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    private static void validateExtensionName(final String extensionName) {
+        if (StringUtils.isBlank(extensionName)) {
+            throw FalconWebException.newAPIException("Extension name is mandatory and shouldn't be blank",
+                    Response.Status.BAD_REQUEST);
+        }
+    }
+
+    private static JSONArray buildEnumerateResult(final List<String> extensions) throws FalconException {
+        JSONArray results = new JSONArray();
+
+        for (String extension : extensions) {
+            String extensionType = AbstractExtension.isExtensionTrusted(extension) ? TRUSTED_EXTENSION
+                    : CUSTOM_EXTENSION;
+
+            JSONObject resultObject = new JSONObject();
+
+            try {
+                resultObject.put(ExtensionElement.NAME.getName(), extension.toLowerCase());
+                resultObject.put(ExtensionElement.Type.getName(), extensionType);
+                resultObject.put(ExtensionElement.DESCRIPTION.getName(), getShortDescription(extension));
+            } catch (JSONException e) {
+                throw new FalconException(e);
+            }
+            results.put(resultObject);
+
+        }
+        return results;
+    }
+
+    private static String getShortDescription(final String extensionName) throws FalconException {
+        String content = getResource(extensionName, extensionName.toLowerCase() + EXTENSION_PROPERTY_JSON_SUFFIX);
+        String description;
+        try {
+            JSONObject jsonObject = new JSONObject(content);
+            description = (String) jsonObject.get(SHORT_DESCRIPTION);
+        } catch (JSONException e) {
+            throw new FalconException(e);
+        }
+        return description;
+    }
+
+    private static String getResource(final String extensionName, final String resourceName) throws FalconException {
```

Comment:
I will move getResource to store but would keep getShortDescription as I don't want to move JSON part to store. Extension store is mainly for operations related to accessing artifacts. Thanks!